cunumeric/deferred.py

```diff
@@ -2588,6 +2588,65 @@ def compute_strides(shape: NdShape) -> tuple[int, ...]:
     def cholesky(self, src: Any, no_tril: bool = False) -> None:
         cholesky(self, src, no_tril)
 
+    @auto_convert([2])
+    def scan(
+        self,
+        op: int,
+        rhs: Any,
+        axis: Optional[int],
+        dtype: Optional[np.dtype[Any]],
+        nan_to_identity: bool,
+    ):
+        # local sum
+        # storage for local sums accessible
+        temp = self.runtime.create_unbound_thunk(
+            dtype=self.dtype, ndim=self.ndim
+        )
+
+        if axis == rhs.ndim - 1:
+            input = rhs
+            output = self
+        else:
+            # swap axes, always performing scan along last axis
+            swapped = rhs.swapaxes(axis, rhs.ndim - 1)
+            input = self.runtime.create_empty_thunk(
+                swapped.shape, dtype=rhs.dtype, inputs=(rhs, swapped)
+            )
+            input.copy(swapped, deep=True)
+            output = self.runtime.create_empty_thunk(
```

Comment:
I don't think you need this empty thunk, as the `input` is already a copy of the original input. You can do the scan in place on the `input` and later copy it back to `self`.

Reply:
1e4cbd2a55ebc72c0ce02a3386c369ac22242e54